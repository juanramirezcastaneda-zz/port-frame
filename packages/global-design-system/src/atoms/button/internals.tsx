import React from 'react';
import { ButtonStyled } from './styled';
import { ButtonProps } from './types';

export function Button({ children, color }: ButtonProps) {
  return <ButtonStyled color={color}>{children}</ButtonStyled>;
}

Button.defaultProps = {
  disabled: false,
  ghost: false,
};
