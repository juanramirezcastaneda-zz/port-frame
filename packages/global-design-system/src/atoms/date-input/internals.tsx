import React from 'react';
import { DateInputProps } from './types';
import { DateInputStyled, DateInputContainerStyled, LabelStyled } from './styled';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

export function DateInput({ className, disabled, isInline, label, setCurrentValue, value }: DateInputProps) {
  const labelComponent = label ? <LabelStyled isInline={isInline}>{label}</LabelStyled> : null;

  return (
    <DateInputStyled className={`${className}`} isInline={isInline}>
      {labelComponent}
      <DateInputContainerStyled>
        <DatePicker className={`form-control`} disabled={disabled} selected={value} onChange={setCurrentValue} />
      </DateInputContainerStyled>
    </DateInputStyled>
  );
}

DateInput.defaultProps = {
  isInline: false,
};
