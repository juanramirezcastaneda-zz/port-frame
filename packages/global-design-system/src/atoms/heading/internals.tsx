import React, { FC } from 'react';
import { HeadingStyled } from './styled';
import { HeadingProps } from './types';

export const Heading: FC<HeadingProps> = ({ children }) => {
  return <HeadingStyled>{children}</HeadingStyled>;
};
