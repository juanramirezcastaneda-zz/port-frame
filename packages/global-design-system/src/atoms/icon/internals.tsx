import React, { FC } from 'react';
import { IconStyled } from './styled';
import { IconProps } from './types';

export const Icon: FC<IconProps> = () => {
  return <IconStyled icon="bars"></IconStyled>;
};
