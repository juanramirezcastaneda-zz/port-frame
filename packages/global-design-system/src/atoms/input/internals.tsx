import React from 'react';
import { InputProps } from './types';
import { LabelStyled, InputStyled, InputComponentStyled, InputContainerStyled } from './styled';

export function Input({ disabled, isInline, label, placeholder, type, value }: InputProps) {
  const labelComponent = label ? <LabelStyled isInline={isInline}>{label}</LabelStyled> : null;

  return (
    <InputStyled isInline={isInline}>
      {labelComponent}
      <InputContainerStyled>
        <InputComponentStyled disabled={disabled} placeholder={placeholder} type={type} value={value} />
      </InputContainerStyled>
    </InputStyled>
  );
}

Input.defaultProps = {
  isInline: false,
};
