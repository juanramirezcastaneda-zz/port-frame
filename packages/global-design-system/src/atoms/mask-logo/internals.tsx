import React from 'react';
import { MaskLogoStyled, MaskLogoSvgStyled, MaskLogoText } from './styled';
import { MaskLogoProps } from './types';

export function MaskLogo({ ...props }: MaskLogoProps) {
  const text = props.orientation === 'Horizontal' ? 'Jsramirezc' : 'The juanramirezcastaneda Site';
  return (
    <MaskLogoStyled>
      <MaskLogoSvgStyled appearance={props.appearance} />
      <MaskLogoText textColor={props.textColor} orientation={props.orientation}>
        {text}
      </MaskLogoText>
    </MaskLogoStyled>
  );
}

MaskLogo.defaultProps = {
  appearance: 'Light',
  orientation: 'Horizontal',
  textColor: 'whiteCore',
};
