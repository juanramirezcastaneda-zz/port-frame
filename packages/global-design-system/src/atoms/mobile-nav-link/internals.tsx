import React from 'react';
import { MobileNavLinkStyled } from './styled';
import { MobileNavLinkProps } from './types';

export function MobileNavLink({ className, children, href, sublink }: MobileNavLinkProps) {
  return (
    <MobileNavLinkStyled className={className} href={href} sublink={sublink}>
      {children}
    </MobileNavLinkStyled>
  );
}
