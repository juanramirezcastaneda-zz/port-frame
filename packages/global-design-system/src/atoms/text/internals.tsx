import React from 'react';
import { TextStyled } from './styled';
import { TextProps } from './types';

export const Text: React.FC<TextProps> = (props) => {
  return <TextStyled {...props} />;
};
