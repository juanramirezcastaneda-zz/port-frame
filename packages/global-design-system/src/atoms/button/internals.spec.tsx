import * as React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './internals';
import { MockThemeProvider } from '~/theme/mock';

describe('Button', () => {
  it('should render a button component as expected', () => {
    const givenLabel = 'fakeLabel';
    const givenIsDisable = false;

    const tree = renderer
      .create(
        <MockThemeProvider>
          <Button disabled={givenIsDisable}>{givenLabel}</Button>
        </MockThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render a button component with ghost prop', () => {
    const givenLabel = 'fakeLabel';
    const givenIsDisable = false;
    const givenIsGhost = true;
    const tree = renderer
      .create(
        <MockThemeProvider>
          <Button disabled={givenIsDisable} ghost={givenIsGhost}>
            {givenLabel}
          </Button>
        </MockThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render a button component with secondary appearance prop', () => {
    const givenLabel = 'fakeLabel';
    const givenIsDisable = false;
    const givenAppearance = 'Secondary';
    const tree = renderer
      .create(
        <MockThemeProvider>
          <Button disabled={givenIsDisable} appearance={givenAppearance}>
            {givenLabel}
          </Button>
        </MockThemeProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
