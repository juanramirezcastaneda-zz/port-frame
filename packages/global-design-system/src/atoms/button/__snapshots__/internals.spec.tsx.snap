// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a button component as expected 1`] = `
<button
  className="sc-bdVaJa bsMTYO"
  disabled={false}
>
  fakeLabel
</button>
`;

exports[`Button should render a button component with ghost prop 1`] = `
<button
  className="sc-bdVaJa bsMTYO"
  disabled={false}
>
  fakeLabel
</button>
`;

exports[`Button should render a button component with secondary appearance prop 1`] = `
<button
  className="sc-bdVaJa eerjhG"
  disabled={false}
>
  fakeLabel
</button>
`;
